kousa/lib/gen/room_chat.ex

```diff
@@ -80,6 +80,48 @@ defmodule Kousa.Gen.RoomChat do
     end
   end
 
+  def handle_cast({:message_deleted, user_id, message_id}, %State{} = state) do
+    last_timestamp = Map.get(state.last_message_map, user_id)
+    {_, seconds, _} = :os.timestamp()
+
+    if not Map.has_key?(state.ban_map, user_id) and
```

Comment:
I don't think you need to check ban_map and I wouldn't worry about throttling deletes